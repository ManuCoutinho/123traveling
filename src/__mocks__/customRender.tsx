import { render } from '@testing-library/react'
import { RenderOptions } from '@testing-library/react/types'
import { ChakraProvider } from '@chakra-ui/react'
import { ReactElement, FC, ReactNode } from 'react'

const MockProviders: FC<{ children: ReactNode }> = ({ children }) => {
	return <ChakraProvider>{children}</ChakraProvider>
}

const customRender = (
	ui: ReactElement,
	options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: MockProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
