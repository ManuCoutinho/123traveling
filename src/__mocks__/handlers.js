import { rest } from 'msw'

export const handlers = [
	rest.get(
		'https://api.unsplash.com/photos/nxkinK7hhOk',
		async (req, res, ctx) => {
			return res(
				ctx.status(200),
				ctx.json({
					full: 'https://images.unsplash.com/photo-1629984557780-4dde2292e245?crop=entropy\u0026cs=srgb\u0026fm=jpg\u0026ixid=MnwzMTA0MzV8MHwxfGFsbHx8fHx8fHx8fDE2ODI1OTYyODU\u0026ixlib=rb-4.0.3\u0026q=85',
					regular:
						'https://images.unsplash.com/photo-1629984557780-4dde2292e245?crop=entropy\u0026cs=tinysrgb\u0026fit=max\u0026fm=jpg\u0026ixid=MnwzMTA0MzV8MHwxfGFsbHx8fHx8fHx8fDE2ODI1OTYyODU\u0026ixlib=rb-4.0.3\u0026q=80\u0026w=1080',
					thumb:
						'https://images.unsplash.com/photo-1629984557780-4dde2292e245?crop=entropy\u0026cs=tinysrgb\u0026fit=max\u0026fm=jpg\u0026ixid=MnwzMTA0MzV8MHwxfGFsbHx8fHx8fHx8fDE2ODI1OTYyODU\u0026ixlib=rb-4.0.3\u0026q=80\u0026w=200',
					small:
						'https://images.unsplash.com/photo-1629984557780-4dde2292e245?crop=entropy\u0026cs=tinysrgb\u0026fit=max\u0026fm=jpg\u0026ixid=MnwzMTA0MzV8MHwxfGFsbHx8fHx8fHx8fDE2ODI1OTYyODU\u0026ixlib=rb-4.0.3\u0026q=80\u0026w=400',
					author: 'Bruno Thethe',
					alt: 'gray concrete bridge over river',
					userLink: 'https://unsplash.com/@bruthethe',
					downloadLocation:
						'https://api.unsplash.com/photos/nxkinK7hhOk/download?ixid=MnwzMTA0MzV8MHwxfGFsbHx8fHx8fHx8fDE2ODI1OTYyODU'
				})
			)
		}
	)
]
