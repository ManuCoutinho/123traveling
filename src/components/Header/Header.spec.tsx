import { renderHook, screen, fireEvent } from '@testing-library/react'
import { useRouter } from 'next/router'
import mockRouter from 'next-router-mock'
import { render } from '__mocks__/customRender'
import { Header } from '.'

jest.mock('next/router', () => require('next-router-mock'))

describe('<Header />', () => {
	beforeEach(() => {
		mockRouter.setCurrentUrl('/home')
	})
	it('should render correctly route ', () => {
		const { result } = renderHook(() => {
			return useRouter()
		})
		expect(result.current).toMatchObject({ pathname: '/home' })
	})
	it('should render menu button in home route ', () => {
		render(<Header />)
		const menu = screen.getByRole('button', {
			hidden: true
		})
		expect(menu).toBeInTheDocument()
		expect(menu).toHaveAttribute('aria-expanded', 'true')
		expect(menu).toHaveAttribute('aria-controls', 'menu')
	})
	it('should render Logo component in Header ', () => {
		render(<Header />)
		expect(screen.getByRole('img', { name: /logo/i })).toBeInTheDocument()
	})
	it('should render menu Toggle button component in Header ', () => {
		render(<Header />)
		const menuClosed = screen.getByRole('button', { hidden: true })
		expect(menuClosed).toBeInTheDocument()
		fireEvent.click(menuClosed)
	})
	it('should match to snapshot ', () => {
		const { container } = render(<Header />)
		expect(container.firstChild).toMatchSnapshot()
	})
})
