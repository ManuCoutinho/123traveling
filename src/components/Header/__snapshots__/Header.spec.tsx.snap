// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match to snapshot  1`] = `
<header
  class="css-1myzxgy"
>
  <div
    class="css-1qiko6f"
  >
    <button
      aria-controls="menu"
      aria-expanded="true"
      aria-hidden="true"
      aria-label="open menu"
      class="chakra-button css-1opyrxp"
      name="menu"
      type="button"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        focusable="false"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M64 384h384v-42.666H64V384zm0-106.666h384v-42.667H64v42.667zM64 128v42.665h384V128H64z"
        />
      </svg>
    </button>
  </div>
  <div
    class="css-16k8qf9"
  >
    <div
      class="css-79elbk"
    >
      <img
        alt="logo com a expressÃ£o Excursionista"
        aria-label="logo"
        data-nimg="1"
        decoding="async"
        height="50"
        loading="lazy"
        role="img"
        src="/assets/logo/logo.svg"
        srcset="/assets/logo/logo.svg 1x, /assets/logo/logo.svg 2x"
        style="color: transparent;"
        width="300"
      />
    </div>
  </div>
</header>
`;
